### PR TITLE
fcsprocket

### PR DESCRIPTION
**Pull request by spontarelliam**
This pull request is for the implementation of a new feature in the Part Design Workbench. It is an ANSI sprocket generator, similar to the gear generator, but for roller chain sprockets.

A discussion of this feature can be found at, https://forum.freecadweb.org/viewtopic.php?f=22&t=44525

A screenshot of the feature is attached.

---
fcsprocket_screenshot

### CODE REVIEW

File: src/Mod/PartDesign/fcsprocket/fcsprocketdialog.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+#   FCGear is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU Library General Public License for more details.
+#
+#   You should have received a copy of the GNU Library General Public
+#   License along with FCGear; if not, write to the Free Software
+#   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307
+
+from PyQt4 import QtGui as qt
```
Comment (vocx-fc) [inline src/Mod/PartDesign/fcsprocket/fcsprocketdialog.py]: Do not use PyQt.

FreeCAD uses PySide. You should import from PySide.

Comment (spontarelliam) [reply src/Mod/PartDesign/fcsprocket/fcsprocketdialog.py]: corrected in fdeef97

**Comment by vocx-fc** [comment]
You should probably squash this pull request into fewer commits, otherwise it's a ton of commits just to add a single tool.

**Comment by spontarelliam** [comment]
Unfortunately this is outside my current git skill level. I'm closing this pull request and opening another with a single commit.

**Comment by vocx-fc** [comment]
Take a look at [Source code management](https://wiki.freecadweb.org/Source_code_management).

The command you need to understand is `git rebase -i`. It's quite powerful once you understand how to squash commits, then edit the commit message. You can rewrite the history of a branch, and make sure everything is neatly organized.

At the end of the mentioned wiki page, there are some videos that teach you about Git. That may be easier to digest than reading just text.

**Pull request closed without merging**